### PR TITLE
feat: custom user agent CognitoUser changes for UI handoff 

### PR DESCRIPTION
**Pull request by erinleigh90**
#### Description of changes
- Creates and exports InternalCognitoUser class from /internals scope with added customUserAgentDetails parameter on public API's that result in service calls
- Sends received customUserAgentDetails through to storage service calls
- Send user agent information from Auth to CognitoUser

NOTES
- To see the changes to InternalCognitoUser after moving the changes over from CognitoUser, see [this comparison](https://github.com/erinleigh90/amplify-js/compare/378bc6b8575ebf8a111bbf0b062f83bf42b425ae...badcffcf7d930d8dfc5435fb28d7f14fcbaf3ec6#diff-e663fb319a13d200f6200fed8357e4975cbe1171a9743d6b8f9720fc852ac83e)

#### Description of how you validated changes
yarn tests pass

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x]  Tests are [changed or added](https://github.com/aws-amplify/amplify-js/blob/main/CONTRIBUTING.md#steps-towards-contributions)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/amazon-cognito-identity-js/src/internals/InternalCognitoUserPool.js [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+ */
+
+import Client from './Client';
+import CognitoUser from './CognitoUser';
+import StorageHelper from './StorageHelper';
+
+const USER_POOL_ID_MAX_LENGTH = 55;
+
+/** @class */
+export default class CognitoUserPool {
```
Comment (elorzafe) [inline packages/amazon-cognito-identity-js/src/internals/InternalCognitoUserPool.js:13]: Should this be called `InternalCognitoUserPool`?

Comment (elorzafe) [reply packages/amazon-cognito-identity-js/src/internals/InternalCognitoUserPool.js:13]: Why `CognitoUserPool` class is not extending this one?

**Review by elorzafe — CHANGES_REQUESTED** [review]
Overall looks good @erinleigh90 

My only concern is `InternalCognitoUserPool` is not `extended` by the external one.

**Review by jimblanc — COMMENTED** [review]
Same concern as Francisco around the `CognitoUserPool` extension, but mostly looks good. It might be a good idea to run the integ tests against this branch just due to the volume of changes (too late now, but `--no-verify` would be useful here to avoid all the prettier changes).

File: packages/amazon-cognito-identity-js/src/internals/InternalCognitoUserPool.js (hunk 1 above)
Comment (erinleigh90) [reply packages/amazon-cognito-identity-js/src/internals/InternalCognitoUserPool.js]: Removed this class, that will be a separate PR for CognitoUserPool changes.

**Review by israx — CHANGES_REQUESTED** [review]
User agent changes look good. I really wanna see this metrics. 

My only concern is about not changing the public facing API types. Although those changes are not breaking. Proposed return values don't align with return values for v6.

File: .circleci/config.yml [hunk 2]
```diff
@@ -1632,6 +1632,7 @@ releasable_branches: &releasable_branches
       - release
       - main
       - next
+      - feat/custom-user-agent-ui/cognito
```
Comment (jimblanc) [inline .circleci/config.yml:1635]: Hum appears this didn't trigger a run in circle CI. Probably need to push feature branch to main repo (can keep the PR branch how it is).

File: packages/amazon-cognito-identity-js/src/Client.js
Comment (erinleigh90) [inline packages/amazon-cognito-identity-js/src/Client.js:1]: Go back through and update jsdocs

**Review by stocaaro — DISMISSED** [review]

**Comment by ashika112** [comment]
This feature was merged in the above linked PRs. Making this draft invalid. Closing this in favor of merged PRs :)

**Pull request closed without merging**